usaspending_api/references/models.py

```diff
@@ -236,8 +237,38 @@ class Location(DataSourceTrackedModel):
     def save(self, *args, **kwargs):
         self.load_country_data()
         self.load_city_county_data()
+        self.fill_missing_state_data()
         super(Location, self).save(*args, **kwargs)
 
+    def state_name_mismatch(self):
+        "True if state_name and state_code not in agreement"
+
+        if (not self.state_name) or (not self.state_code):
+            return False
+        return (self.state_name != code_to_state.get(self.state_code))
+
+    def is_us_state(self):
+        "Is this unambiguously a US state?"
+
+        if self.state_name_mismatch():
+            return False
+
+        if (self.state_code in code_to_state) or (self.state_name in state_to_code):
+            return True
+
+    def fill_missing_state_data(self):
+        """Fills in blank US state names or codes from its counterpart"""
+
+        if self.state_code and self.state_name:
+            return
+        if self.state_name_mismatch():
```

Comment:
State mismatch cant be true since it already returned if state_code and state_name both exist